src/dbnode/storage/index/postings_list_cache_lru.go

```diff
@@ -0,0 +1,237 @@
+	if size <= 0 {
+		return nil, errors.New("Must provide a positive size")
+	}
+
+	return &postingsListLRU{
+		size:      size,
+		evictList: list.New(),
+		items:     make(map[uuid.Array]map[patternAndPatternType]*list.Element),
+	}, nil
+}
+
+// Add adds a value to the cache. Returns true if an eviction occurred.
+func (c *postingsListLRU) Add(
+	segmentUUID uuid.UUID,
+	pattern string,
+	patternType PatternType,
+	pl postings.List,
+) (evicted bool) {
+	// Check for existing item.
+	uuidArray := segmentUUID.Array()
+	if uuidEntries, ok := c.items[uuidArray]; ok {
+		key := newPatternAndPatternType(pattern, patternType)
+		if ent, ok := uuidEntries[key]; ok {
+			// If it already exists, just move it to the front. This avoids storing
+			// the same item in the LRU twice which is important because the maps
+			// can only point to one entry at a time and we use them for purges. Also,
+			// it saves space by avoiding storing duplicate values.
+			c.evictList.MoveToFront(ent)
+			ent.Value.(*entry).postingsList = pl
+			return false
+		}
+	}
+
+	// Add new item.
+	var (
+		ent = &entry{
+			uuid:         segmentUUID,
+			pattern:      pattern,
+			patternType:  patternType,
+			postingsList: pl,
+		}
+		entry = c.evictList.PushFront(ent)
+		key   = newPatternAndPatternType(pattern, patternType)
+	)
+	if patterns, ok := c.items[uuidArray]; ok {
+		patterns[key] = entry
+	} else {
+		c.items[uuidArray] = map[patternAndPatternType]*list.Element{
+			key: entry,
+		}
+	}
+
+	evict := c.evictList.Len() > c.size
+	// Verify size not exceeded.
+	if evict {
+		c.removeOldest()
+	}
+	return evict
+}
+
+// Get looks up a key's value from the cache.
+func (c *postingsListLRU) Get(
+	segmentUUID uuid.UUID,
+	pattern string,
+	patternType PatternType,
+) (postings.List, bool) {
+	uuidArray := segmentUUID.Array()
+	if uuidEntries, ok := c.items[uuidArray]; ok {
+		key := newPatternAndPatternType(pattern, patternType)
+		if ent, ok := uuidEntries[key]; ok {
+			c.evictList.MoveToFront(ent)
+			return ent.Value.(*entry).postingsList, true
+		}
+	}
+
+	return nil, false
+}
+
+// Remove removes the provided key from the cache, returning if the
+// key was contained.
+func (c *postingsListLRU) Remove(
+	segmentUUID uuid.UUID,
+	pattern string,
+	patternType PatternType,
+) bool {
+	uuidArray := segmentUUID.Array()
+	if uuidEntries, ok := c.items[uuidArray]; ok {
+		key := newPatternAndPatternType(pattern, patternType)
+		if ent, ok := uuidEntries[key]; ok {
+			c.removeElement(ent)
+			return true
+		}
+	}
+
+	return false
+}
+
+func (c *postingsListLRU) PurgeSegment(segmentUUID uuid.UUID) {
+	if uuidEntries, ok := c.items[segmentUUID.Array()]; ok {
+		for _, ent := range uuidEntries {
+			c.removeElement(ent)
+		}
+	}
+}
+
+// Len returns the number of items in the cache.
+func (c *postingsListLRU) Len() int {
+	return c.evictList.Len()
+}
+
+// Keys returns a slice of the keys in the cache, from oldest to newest. Mostly
```

Comment:
if it's only used for testing, can you move it to the `_test.go` file

Reply:
done